lib/common/filters/retrypolicyfilter.js

```diff
@@ -217,7 +217,7 @@ RetryPolicyFilter._shouldRetryOnError = function (statusCode, requestOptions) {
         } else {
           retryInfo.retryable = false;
         }
-      } else if (retryInfo.retryable && statusCode >= 500 && statusCode < 600) {
+      } else if (retryInfo.retryable !== false && statusCode >= 500 && statusCode < 600) {
```

Comment:
Can we totally remove the `retryInfo.retryable` from if condition? Could `retryInfo.retryable` be `false`?

```suggestion
      } else if (statusCode >= 500 && statusCode < 600) {
```

Reply:
Does this fit that scenario?
https://github.com/Azure/azure-storage-node/blob/master/lib/services/blob/blobservice.core.js#L5405

`// None of the below cases should be retried. So set the error in every case so the retry policy filter handle knows that it shouldn't be retried.
  if (options.disableContentMD5Validation === false && options.useTransactionalMD5 === true && azureutil.objectIsNull(storedMD5)) {
    responseObject.error = new StorageError(SR.MD5_NOT_PRESENT_ERROR);
    responseObject.retryable = false;
  }`

Reply:
Not exactly, the `responseObject.retryable` degined in above code is not same with `retryinfo.retryable`.

There are 2 `retryable` properties actually.

The 1st `retryable` property is defined in above code. `responseObject` in above code will pass to retrypolicyfilter as `requestOptions`. After a `next` call, it "becomes" `returnObject`. This line of code checks `returnObject.retryable`:

https://github.com/Azure/azure-storage-node/blob/4df4cbd12c8a4d7b4aff9945cb8ca5a99fd97541/lib/common/filters/retrypolicyfilter.js#L133

The 2nd `retryinfo.retryable` is initialized in `retrypolicyfilter` here. It's part of `retryContext`.

https://github.com/Azure/azure-storage-node/blob/4df4cbd12c8a4d7b4aff9945cb8ca5a99fd97541/lib/common/filters/retrypolicyfilter.js#L81

Reply:
Thanks for clarifying. Then it makes sense to remove the `retryInfo.retryable` condition like you suggested.